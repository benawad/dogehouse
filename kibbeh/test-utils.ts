import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import "mutationobserver-shim";

const customRender = (ui: any, options = {}) =>
  render(ui, {
    ...options,
  });

export * from "@testing-library/react";

export { customRender as render };
