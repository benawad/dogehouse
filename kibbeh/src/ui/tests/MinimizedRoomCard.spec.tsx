import React from "react";

import { render } from "../../../test-utils";
import {
  MinimizedRoomCard,
  MinimizedRoomCardProps,
} from "../MinimizedRoomCard";

const defaultProps: MinimizedRoomCardProps = {
  room: {
    name: "",
    roomStartedAt: new Date(),
    speakers: ["user1", "user2", "user3"],
    myself: {
      isDeafened: false,
      isMuted: false,
      isSpeaker: true,
      switchDeafened: () => {},
      leave: () => {},
      switchMuted: () => {},
    },
  },
};

describe("MinimizedRoomCard", () => {
  it("should render correctly", () => {
    const { getByTestId } = render(<MinimizedRoomCard {...defaultProps} />);
    const component = getByTestId("minimized-room-card");

    expect(component).toBeVisible();
  });

  it("should render mic icon if myself isn't muted", () => {
    const { getByTestId } = render(<MinimizedRoomCard {...defaultProps} />);
    const component = getByTestId("mic-on");

    expect(component).toBeVisible();
  });

  it("should render mic-off icon if myself is muted", () => {
    const props: MinimizedRoomCardProps = {
      ...defaultProps,
      room: {
        ...defaultProps.room,
        myself: {
          ...defaultProps.room.myself,
          isMuted: true,
        },
      },
    };
    const { getByTestId } = render(<MinimizedRoomCard {...props} />);
    const component = getByTestId("mic-off");

    expect(component).toBeVisible();
  });

  it("should match snapshot", () => {
    const { getByTestId } = render(<MinimizedRoomCard {...defaultProps} />);
    const component = getByTestId("minimized-room-card");

    expect(component).toMatchSnapshot();
  });
});
