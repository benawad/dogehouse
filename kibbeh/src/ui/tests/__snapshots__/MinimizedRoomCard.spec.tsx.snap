// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinimizedRoomCard should match snapshot 1`] = `
<div
  class="bg-primary-800 border border-accent rounded-lg p-4 gap-4 grid max-w-md w-full"
  data-testid="minimized-room-card"
>
  <div
    class="gap-1 grid"
  >
    <h4
      class="text-primary-100 break-all overflow-hidden"
    />
    <div
      class="text-primary-300 overflow-ellipsis overflow-hidden w-auto"
    >
      user1, user2, user3
    </div>
    <div
      class="text-accent"
    >
      components.bottomVoiceControl.speaker
       
      Â· 
      00:00
    </div>
  </div>
  <div
    class="flex flex-row"
  >
    <div
      class="grid grid-cols-3 gap-2"
    >
      <button
        class="flex bg-primary-700 transition duration-200 ease-in-out hover:bg-primary-600 h-6 w-6 cursor-pointer justify-center items-center rounded 
        bg-accent hover:bg-accent-hover text-button"
        data-testid="mute"
      >
        <svg
          data-testid="mic-on"
          fill="currentColor"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 11.205a3.2 3.2 0 003.196-3.197V3.213C11.196 1.442 9.77 0 8.016 0a.746.746 0 00-.167.02 3.201 3.201 0 00-3.046 3.193v4.795a3.2 3.2 0 003.196 3.197z"
          />
          <path
            d="M7.2 14.346V16H8.8v-1.654c3.148-.395 5.594-3.083 5.594-6.338h-1.598A4.8 4.8 0 018 12.803a4.8 4.8 0 01-4.795-4.795H1.606c0 3.255 2.447 5.943 5.595 6.338z"
          />
        </svg>
      </button>
      <button
        class="flex bg-primary-700  hover:bg-primary-600 h-6 w-6 cursor-pointer justify-center items-center rounded text-primary-100
        "
        data-testid="deafen"
      >
        <svg
          data-testid="headphone-on"
          fill="currentColor"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13.5414 2.33786C12.0154 0.852143 9.99572 0 8 0C6.00429 0 3.98464 0.852143 2.45857 2.33786C0.873214 3.88107 0 5.89286 0 8C0 8.9525 0.3125 10.1818 1.17429 12.4839C2.03607 14.7861 3.75 15.75 4.47393 15.9075C4.68143 15.9529 4.92464 16 5.14286 16C5.51887 16.0003 5.88836 15.9018 6.21429 15.7143L6.71429 15.4286C7.2525 15.1136 7.40964 14.4239 7.1 13.8804L3.99571 8.43143C3.92174 8.30104 3.82264 8.18663 3.70414 8.09481C3.58564 8.003 3.4501 7.93561 3.30537 7.89654C3.16064 7.85748 3.0096 7.84752 2.86099 7.86724C2.71239 7.88696 2.56917 7.93597 2.43964 8.01143L1.95071 8.29714C1.7628 8.40723 1.59395 8.54701 1.45071 8.71107C1.43294 8.73193 1.40953 8.74724 1.38329 8.75516C1.35706 8.76309 1.32909 8.7633 1.30274 8.75577C1.27638 8.74824 1.25275 8.73329 1.23466 8.7127C1.21657 8.69211 1.20479 8.66675 1.20071 8.63964C1.16472 8.42827 1.14537 8.2144 1.14286 8C1.14286 6.20321 1.89286 4.48286 3.25571 3.15679C4.57143 1.87679 6.29964 1.14286 8 1.14286C9.70036 1.14286 11.4286 1.87679 12.7443 3.15679C14.1071 4.48286 14.8571 6.20321 14.8571 8C14.8539 8.21446 14.8339 8.42833 14.7971 8.63964C14.7931 8.66675 14.7813 8.69211 14.7632 8.7127C14.7451 8.73329 14.7215 8.74824 14.6951 8.75577C14.6688 8.7633 14.6408 8.76309 14.6146 8.75516C14.5883 8.74724 14.5649 8.73193 14.5471 8.71107C14.4039 8.54701 14.2351 8.40723 14.0471 8.29714L13.5582 8.01143C13.4287 7.93597 13.2855 7.88696 13.1369 7.86724C12.9883 7.84752 12.8372 7.85748 12.6925 7.89654C12.5478 7.93561 12.4122 8.003 12.2937 8.09481C12.1752 8.18663 12.0761 8.30104 12.0021 8.43143L8.9 13.8804C8.59036 14.4239 8.7475 15.1136 9.28572 15.4286L9.78572 15.7143C10.1116 15.9018 10.4811 16.0003 10.8571 16C11.0754 16 11.3186 15.9529 11.5261 15.9075C12.25 15.75 13.9643 14.7857 14.8257 12.4839C15.6871 10.1821 16 8.9525 16 8C16 5.89286 15.1268 3.88107 13.5414 2.33786Z"
          />
        </svg>
      </button>
      <button
        class="flex bg-primary-700 transition duration-200 ease-in-out hover:bg-primary-600 h-6 w-6 cursor-pointer justify-center items-center rounded text-primary-100
        "
        data-testid="boxed-icon"
      >
        <svg
          fill="currentColor"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M2.797 1.559h.934a.78.78 0 000-1.559H.747A.764.764 0 000 .78v3.118a.747.747 0 101.492 0v-1.5l3.144 3.328a.755.755 0 101.066-1.069L2.797 1.56zm10.38.01h-.945a.785.785 0 010-1.569h3.014a.768.768 0 01.753.785v3.143a.755.755 0 11-1.507 0v-1.51l-3.177 3.353a.762.762 0 11-1.078-1.079l2.94-3.123zM2.796 14.441h.934a.78.78 0 010 1.559H.747A.764.764 0 010 15.22v-3.118a.747.747 0 111.492 0v1.5l3.144-3.328a.755.755 0 111.066 1.069L2.797 14.44zm10.38-.01h-.945a.785.785 0 000 1.569h3.014a.767.767 0 00.753-.784v-3.143a.755.755 0 10-1.507 0v1.508l-3.177-3.352a.762.762 0 10-1.078 1.079l2.94 3.123z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
    <button
      class="flex outline-none focus:ring-4 focus:ring-primary-300 py-2 px-6 text-sm rounded-lg transition duration-200 ease-in-out text-button bg-primary-700 hover:bg-primary-600 disabled:text-primary-300 font-bold flex items-center justify-center flex-grow ml-4"
      data-testid="button"
    >
      <span
        class="flex items-center"
      >
        components.bottomVoiceControl.leave
      </span>
    </button>
  </div>
</div>
`;
